lavc : libyami dec/enc reuse template thread function in libyami_inte…

popkartyeah opened the pull request:
…rnal.h

libyami dec/enc reuse template thread function in libyami_internal.h

Commit:
lavc : libyami dec/enc reuse template thread function in libyami_internal.h

libyami dec/enc reuse template thread function in libyami_internal.h

Commit:
lavc : libyami dec/enc reuse template thread function in
libyami_internal.h

libyami dec/enc reuse template thread function in libyami_internal.h

mypopydev commented on libavcodec/libyami_dec.cpp:
```diff
@@ -304,8 +242,8 @@ static int yami_dec_init(AVCodecContext *avctx)
     }
-    s->in_queue = new std::deque<VideoDecodeBuffer*>;
 
+//    s->avctx = avctx;
```
Delete un-used code

Commit:
lavc : delete unused comments.

delete unused comments.

Commit:
Merge branch 'change_thread' of https://github.com/popkartyeah/ffmpeg_libyami into change_thread

Commit:
fate : add vp9 decoder fate case.

add vp9 decoder fate case.

Commit:
Merge branch 'master' of https://github.com/01org/ffmpeg_libyami into change_thread

Conflicts:
	libavcodec/libyami_dec.cpp
	libavcodec/libyami_enc.cpp

Commit:
lavc : fix vcl config_buffer should init config_buffer height width issue.

fix vcl config_buffer should init config_buffer height width issue.

Commit:
cfg : moddify configure file mode.

moddify configure file mode.

Commit:
fate : clean libyami VC1 test case

delete the msvideo1 mpeg4 test case in libyami VC1 fate

Commit:
Merge branch 'master' of https://github.com/01org/ffmpeg_libyami into change_thread

Commit:
lavc : fix libyami_h264 stream loop cannot work issue.

1.fix libyami_h264 stream loop cannot work issue, add new status YAMI_THREAD_FLUSH_OUT , when the the stream get eos the thread flush out all frames will change to this state.
2.add ff_yami_set_stream_run function to set the thread status YAMI_THREAD_RUNNING.
3.call the ff_yami_thread_close to cancel the thread.
